Commands/gemini3.py

```
@@ -126,33 +237,38 @@ def reply_with_grounded_gemini(self, message):
-        if duck_urls:
-            self.send_privmsg(cmd.channel, f"📝 Source(s): {' | '.join(duck_urls)}")
+        if valid_urls:
+            self.send_privmsg(cmd.channel, f"📝 Source(s): {' | '.join(valid_urls)}")
+
     except Exception as e:
         print(f"[Error] {e}")
         self.send_privmsg(cmd.channel, f"Failed to send a response. Please try again later")
-
+    
+    finally:
+        return
```

Ruff (0.11.9)

`return` inside `finally` blocks cause exceptions to be silenced

(B012)